cmake/common.cmake

```diff
@@ -115,6 +116,17 @@ macro(spl_create_component)
     # Collect all component names for later usage (e.g., in an extension)
     list(APPEND COMPONENT_NAMES ${component_name})
     set(COMPONENT_NAMES ${COMPONENT_NAMES} PARENT_SCOPE)
+    
+    # Collect all component information for sphinx documentation
+    set(_component_info "{

```

Comment:
We need to keep track of all components and their information to be able to generate the variant reports. For the variants reports, one need to loop over all components and generate component variant specific targets.